spec/prog/kubernetes/kubernetes_cluster_nexus_spec.rb

```diff
@@ -5,51 +5,57 @@
 RSpec.describe Prog::Kubernetes::KubernetesClusterNexus do
   subject(:nx) { described_class.new(Strand.new(id: "8148ebdf-66b8-8ed0-9c2f-8cfe93f5aa77")) }
 
-  let(:project) { Project.create_with_id(name: "default") }
-  let(:subnet) { PrivateSubnet.create_with_id(net6: "0::0", net4: "127.0.0.1", name: "x", location: "x", project_id: project.id) }
+  let(:customer_project) { Project.create_with_id(name: "default") }
 
   let(:kubernetes_cluster) {
+    subnet = Prog::Vnet::SubnetNexus.assemble(
+      Config.kubernetes_service_project_id,
+      location: "hetzner-fsn1",
+      ipv4_range: "10.10.0.0/16"
+    ).subject
+
     kc = KubernetesCluster.create_with_id(
       name: "k8scluster",
       kubernetes_version: "v1.32",
       cp_node_count: 3,
       private_subnet_id: subnet.id,
       location: "hetzner-fsn1",
-      project_id: project.id
+      project_id: customer_project.id
     )
 
-    lb = LoadBalancer.create_with_id(private_subnet_id: subnet.id, name: "somelb", src_port: 123, dst_port: 456, health_check_endpoint: "/foo", project_id: project.id)
+    lb = LoadBalancer.create_with_id(private_subnet_id: subnet.id, name: "somelb", src_port: 123, dst_port: 456, health_check_endpoint: "/foo", project_id: Config.kubernetes_service_project_id)
     kc.add_cp_vm(create_vm)
     kc.add_cp_vm(create_vm)
     kc.update(api_server_lb_id: lb.id)
     kc
   }
 
   before do
+    allow(Config).to receive(:kubernetes_service_project_id).and_return(Project.create(name: "UbicloudKubernetesService").id)
     allow(nx).to receive(:kubernetes_cluster).and_return(kubernetes_cluster)
   end
 
   describe ".assemble" do
     it "validates input" do
       expect {
-        described_class.assemble(project_id: SecureRandom.uuid, name: "k8stest", kubernetes_version: "v1.32", location: "hetzner-fsn1", cp_node_count: 3, private_subnet_id: subnet.id)
+        described_class.assemble(project_id: SecureRandom.uuid, name: "k8stest", kubernetes_version: "v1.32", location: "hetzner-fsn1", cp_node_count: 3)
```

Comment:
You might use hardcoded project ID

https://github.com/ubicloud/ubicloud/pull/799